plugins/chat/assets/javascripts/discourse/components/chat-thread-title-prompt.js

```diff
@@ -24,41 +23,30 @@ export default class ShowThreadTitlePrompt {
   constructor(owner, thread) {
     setOwner(this, owner);
     this.thread = thread;
-    this.createThreadMembership();
   }
 
   get membership() {
     return this.thread.currentUserMembership;
   }
 
-  @action
-  createThreadMembership() {
-    if (this.membership || !this.currentUser.admin) {
-      return;
-    }
-
-    this.thread.currentUserMembership = UserChatThreadMembership.create({
-      notification_level: NotificationLevels.TRACKING,
-      last_read_message_id: this.thread.lastMessageId,
-      thread_title_prompt_seen: false,
-    });
-  }
-
   @action
   async updateThreadTitlePrompt() {
-    if (!this.membership) {
-      return;
-    }
-
     try {
-      await this.chatApi.updateCurrentUserThreadTitlePrompt(
+      const result = await this.chatApi.updateCurrentUserThreadTitlePrompt(
         this.thread.channel.id,
         this.thread.id
       );
+
+      this.thread.currentUserMembership = UserChatThreadMembership.create(
```

Comment:
Just rely on the fact that mark thread prompt as seen will return the membership, and we can update/set the `currentUserMembership` using this.

Reply:
I think this should work, as if there is no membership then it prompt seen would be false anyway.